### PR TITLE
bump maven-bundle-plugin

### PR DESCRIPTION
**Pull request by cdietrich**
bump maven-bundle-plugin
will help to solve the problems reported in

https://bugs.eclipse.org/bugs/show_bug.cgi?id=580280
https://github.com/eclipse-pde/eclipse.pde/issues/130
https://github.com/eclipse/tycho/discussions/1084

### CODE REVIEW

**Comment by cdietrich** [comment]
is currently blocked by merge and release of https://github.com/apache/felix-dev/pull/160

File: releng/maven-plugins/ebr-maven-plugin/pom.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <dependency>
       <groupId>org.apache.felix</groupId>
       <artifactId>maven-bundle-plugin</artifactId>
-      <version>5.1.4</version>
+      <version>5.1.7</version>
```
Comment (cdietrich) [inline releng/maven-plugins/ebr-maven-plugin/pom.xml:43]: a workaround might be to use the old version and override the transitive dependency

**Comment by HannesWell** [comment]
I don't know exactly what the maven-bundle-plugin is used for, but can't we just migrate to the `bnd-maven-plugin`?
Felix's maven-bundle-plugin uses that under the hood anyways and using it directly would save us the intermediate step.

**Comment by cdietrich** [comment]
i dont know if the classes we inherit from is there
e.g. https://github.com/eclipse/ebr/blob/d6b406d16e7e31fb055a9fab0037b52b34b3a0c4/releng/maven-plugins/ebr-maven-plugin/src/main/java/org/eclipse/ebr/maven/BundleMojo.java#L65

**Comment by guw** [comment]
Can you try overriding the bnd dependency of the plug-in definition?
```
  <dependency>
    <groupId>biz.aQute.bnd</groupId>
    <artifactId>biz.aQute.bndlib</artifactId>
    <version>6.3.1</version>
  </dependency>
```

**Comment by cdietrich** [comment]
alternative would be https://github.com/eclipse/ebr/pull/57

**Comment by cdietrich** [comment]
the 5.1.7 version of the maven-bundle-plugin is published now

**Comment by HannesWell** [comment]
> the 5.1.7 version of the maven-bundle-plugin is published now

Great!

Can we then get a new version of the flawed xml.resolver bundle?

**Comment by cdietrich** [comment]
unfortunately i dont know how orbit decides to build and pick a new artifact
 maybe @jonahgraham or @rgrunber  can help on that

**Comment by rgrunber** [comment]
orbit-recipes appears to be using a SNAPSHOT version of 1.4.0, https://git.eclipse.org/c/orbit/orbit-recipes.git/tree/releng/mavenparent/pom.xml#n67 . So as long as a build of the ebr-maven-plugin happened for this commit, a simple rebuild of orbit-recipes should fix the problem going forward.

Looking at https://repo.eclipse.org/content/repositories/ebr-snapshots/org/eclipse/ebr/ebr-maven-plugin/1.4.0-SNAPSHOT/ ( https://ci.eclipse.org/ebr/job/ebr-maven-plugins/ ), I think this commit is now in the snapshots.

**Comment by iloveeclipse** [comment]
> a simple rebuild of orbit-recipes should fix the problem going forward

Sorry for a willy question : how one can trigger such "simple rebuild"? Could you do this?

**Comment by rgrunber** [comment]
> > a simple rebuild of orbit-recipes should fix the problem going forward
> 
> Sorry for a silly question : how one can trigger such "simple rebuild"? Could you do this?

Any committer to the Orbit project should have permissions to trigger the job. I've triggered https://ci.eclipse.org/orbit/job/orbit-recipes/413/ .

```
16:35:57  [INFO] Downloading from ebr-snapshots: https://repo.eclipse.org/content/repositories/ebr-snapshots/org/eclipse/ebr/ebr-maven-plugin/1.4.0-SNAPSHOT/ebr-maven-plugin-1.4.0-20220714.195314-41.jar
...
...
16:35:57  [INFO] Downloading from ebr-releases: https://repo.eclipse.org/content/repositories/ebr-releases/org/apache/felix/maven-bundle-plugin/5.1.7/maven-bundle-plugin-5.1.7.jar
```

**Comment by guw** [comment]
EBR build is here: https://ci.eclipse.org/ebr/job/ebr-maven-plugins/523/

**Comment by cdietrich** [comment]
this is what i meant with 
> unfortunately i dont know how orbit (build) decides to build and pick a new artifact

**Comment by iloveeclipse** [comment]
@nitind : as owner of `org.apache.xml.resolver` on Orbit, do **you** need to do something to get updated version on Orbit? It is required for webtools, see https://bugs.eclipse.org/bugs/show_bug.cgi?id=580383

**Comment by jonahgraham** [comment]
I think you'll need to force the qualifier to update. See https://git.eclipse.org/r/c/orbit/orbit-recipes/+/193191 for an example.

Otherwise we end up with the same qualified version and potentially different content.

**Comment by cdietrich** [comment]
created https://git.eclipse.org/r/c/orbit/orbit-recipes/+/194732

**Pull request merged**